### PR TITLE
Migrating to a more local-friendly setup

### PR DESCRIPTION
**Pull request by stjohnjohnson**
- Uses a config manager
- Can set settings via a config.yaml OR environment variables
- Simplifies server startup

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,6 +11,9 @@ RUN npm install --production
 # Copy everything else
 COPY . /usr/src/app
 
+# Setup configuration folder
+RUN ln -s /usr/src/app/config /config
```
Comment (nkatzman) [inline Dockerfile]: why does it need to be linked to /config ?

Comment (stjohnjohnson) [inline Dockerfile]: easier for users to add local.yaml

File: README.md [hunk 2]
```diff
@@ -1,61 +1,59 @@
...
-    port: 8666
-}, (error, server) {
-    if (error) {
-        return console.error(error);
-    }
-    console.log('Server running at ', server.info.uri);
-});
+## Configuration
+
+Screwdriver already [defaults most configuration](config/default.yaml), but you can override any of it by either be a `local.yaml` or environment variables.
```
Comment (tkyi) [inline README.md]: but you can override **defaults using a** `local.yaml` or environment variables.

File: bin/server [hunk 3]
```diff
@@ -2,27 +2,34 @@
...
+}
+if (config.has('executor.k8s.token')) {
+    process.env.K8S_TOKEN = config.get('executor.k8s.token');
+}
+
+// Setup Login
+const loginConfig = config.get('login');
+
+// Setup httpd
+const httpdConfig = config.get('httpd');
```
Comment (nkatzman) [inline bin/server]: where are you using this?

**Comment by jer** [comment]
687474703a2f2f692e696d6775722e636f6d2f334846754465342e676966

**Comment by nkatzman** [comment]
👍

**Pull request merged**